pkgs/test_core/lib/src/runner/vm/test_compiler.dart

```diff
@@ -159,13 +159,25 @@ class _TestCompilerForLanguageVersion {
     final platformDill = 'lib/_internal/vm_platform_strong.dill';
     final sdkRoot =
         p.relative(p.dirname(p.dirname(Platform.resolvedExecutable)));
+    final packageConfigUriAwaited = await packageConfigUri;
+
+    // If we have native assets for the host os in JIT mode, they are here.
+    Uri? nativeAssetsYaml;
+    if (enabledExperiments.contains('native-assets')) {
+      nativeAssetsYaml = packageConfigUriAwaited.resolve('native_assets.yaml');
+      if (!await File.fromUri(nativeAssetsYaml).exists()) {
+        nativeAssetsYaml = null;
+      }
+    }
+
     var client = _frontendServerClient = await FrontendServerClient.start(
       testUri.toString(),
       _outputDill.path,
       platformDill,
       enabledExperiments: enabledExperiments,
       sdkRoot: sdkRoot,
-      packagesJson: (await packageConfigUri).toFilePath(),
+      packagesJson: packageConfigUriAwaited.toFilePath(),
+      nativeAssets: nativeAssetsYaml?.toFilePath(),
```

Comment:
It's unlikely this argument existed in the `2.1.0` of `frontend_server_client`. We need to bump the lower bound of the constraint to whatever version adds support for this argument.

Reply:
We need to bump it to `3.2.0`https://pub.dev/documentation/frontend_server_client/latest/frontend_server_client/FrontendServerClient/start.html

Reply:
Good catch @natebosch !